pkg/server/options/genericcontrolplane.go

```diff
@@ -0,0 +1,112 @@
+package options
+
+/*
+   Copyright 2022 The KCP Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"fmt"
+
+	cliflag "k8s.io/component-base/cli/flag"
+	_ "k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/options"
+)
+
+//
+// Mostly copied from k8s.io/kubernetes/cmd/kube-apiserver/app/options
+//
+// TODO: move all back into genericcontrolplane
+//
+
+type ServerRunOptions struct {
+	options.ServerRunOptions
+}
+
+// Flags returns flags for a specific APIServer by section name
+func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
+	// TODO: move all these back into genericcontrolplane
+	s.GenericServerRunOptions.AddUniversalFlags(fss.FlagSet("generic"))
```

Comment:
Would prefer fewer flags rather than more being exposed to users right now.  150+ flags is too many - I might prefer an allowlist mindset (if that hasn't already been thought of and i just missed it in review)

Reply:
agree, especially for these generic one. Most look like random one-offs.